### DISCUSSION TITLE
On MacBook Air M1 it is extremely slow, unusable!

### DISCUSSION THREAD

**Discussion by eziologico** [utmapp/UTM#2533] [UTM Help (macOS)]
I have tried to create some x86_64 virtual machines with Linux and Windows but it is impossible to use them as they are very slow.
I downloaded some ready-made virtual machines from the UTM site but they have the same problem.
In addition, the CPU heats up a lot reaching over 70 ° C

macOS Big Sur v11.3.1
UTM v2.0.28 (28)

**Comment by osy** [comment]
Yes that’s expected as x86_64 is being emulated.

**Reply by thedocbwarren** [reply]
Noticed IO is super slow.  Not so much interacting with UI.  CLIs are quick though.  Not sure if it's the CPU emulation that is slow or IO drivers.  Other the flip side Docker uses QEMU for emulating x86 images.  They do not run painfully slow.  Why does this?

**Reply by Atemu** [reply]
I'm pretty sure Docker uses user-mode emulation where the OS is native aarch64 but the app binaries are run via qemu-user; similar to how Rosetta works. This can save a lot of cycles because the kernel is native.

It's ultimately still the same qemu emulation though. You probably don't notice as much as the workloads of your local docker env is likely to be orders of magnitudes lighter than a typical desktop system.

**Comment by GuizmoFR** [comment]
I also have this issues and I notice that UTM run the VM on 1 cpu only on. my MacBook Air M1. I well configure my vm with a mx of 4 cpu but even Windows detect only 1 CPU.
Is there a way to add multicore for Win10 VM ? and the UTM use more than 1 core ?

**Reply by Jakersnell** [reply]
force multicore in vm settings, but its unstable.

**Comment by sscarduzio** [comment]
Hello from 2022 (UTM 3.0.3 beta), in my top specs M1 Max (macbook pro 14" late 2021) Ubuntu Desktop for amd64 is equally unusably slow with default settings.

I reconfigured the VM to force 4 cores and 4GB memory, now it's aaalmost usable. But programs still open slowly, including the file manager. Once they are open it's kinda ok. I suspect there's some high latency in the I/O? Not sure how to measure this though. 

Any suggestion on how to speed it up?

**Reply by osy** [reply]
That’s expected. You’re emulating an entire CPU. This is not optimized like apple’s Rosetta 2.

**Reply by sscarduzio** [reply]
CPU is not the bottleneck, Memory is not the bottleneck, so I disagree. I think it's the million abstraction layers between disk IO in the VM and the disk image in the host. 

I'm evaluating if it's possible creating a disk partition on the host and somehow let qemu handle it directly (as opposed to the image file).

**Reply by osy** [reply]
One way to check is to use an arm64 vm and see if you get similar io issues.

**Reply by DJ-IRL** [reply]
i have a vm using arm64 (aarch64) UTM on mac m1 '21, speed and usability is damn near native. really fast and works fine. 

trying to emulate x86 architecture so I can run some binaries my school gave me, but that VM is entirely unusable. apps opening and mouse clicks both take actual minutes to complete. 

any solution for running x86 binary files on the aarch64 architecture somehow? any direction is appreciated :D

**Reply by osy** [reply]
You can try force multicore with a higher core count at the cost of a chance of incorrect emulation.

**Reply by DJ-IRL** [reply]
okay running ubuntu x86 rn from utm gallery, only changes I've made are forcing multicore and I increased the memory from 512 to 4096. 

I have to admit it's faster. but not by enough to be usable. firefox still took 90 seconds to open. 

also, i didnt realize i was in the official repo lol. nice to meet ya, beautiful software overall. just having an issue w custom binaries from my professor

**Reply by sullof** [reply]
@osy It makes sense, but you should alert people about that before that they buy it.

**Reply by osy** [reply]
I believe it says multiple times in the App Store description that emulation will be slower then virtualization and there’s no gpu acceleration.

**Reply by naisanzaa** [reply]
This. If even the M1 Max is suffering, hopefully once that's solved, this will become usable for all M1's.

**Reply by Atemu** [reply]
Are those apps snaps? If yes, that startup time is (somewhat) expected.

Try using non-snap apps and/or a distro which doesn't employ snaps (basically anything but Ubuntu).

**Comment by nonarkitten** [comment]
Just adding my $0.02, running the Windows XP is very snappy, but likes to crash. Not even a BSOD, just back to BIOS boot. But performance-wise the thing runs like hot buttered toast. Windows 7 runs like hot garbage. Opening Windows Explorer can take minutes some times and gets exponentially worse the more you're doing on it. Upping the RAM and CPU core count had little impact.

**Comment by pavitrakumar78** [comment]
can anyone comment on the current performance of running x86 ubuntu on M1/M2 - no GUI, just cmd/headless?

**Reply by nonarkitten** [reply]
Okay, so using the GUI is better than text mode. The GUI can be accelerated, though it's possibly buggy and labelled as "experimental" I'm not sure I've seen it mess up. I've had some random crashes of UTM, but that could have been anything and happens on some systems without this feature. In GUI mode, opening a terminal and using it is great.

Side note, boy it would be nice to get the OpenGL acceleration on other platforms. cough Windows cough

Text mode does not have any acceleration and sometimes is so slow I can watch lines being drawn like the old fill-routines on Deluxe Paint on the Amiga. Like wow, scrolling text shouldn't take 1-2 seconds. Not sure why this happens, but it's pretty consistently a bad experience. Maybe Linux likes using graphical modes for text (doubt it) but text should be so fast on a modern Mac as to be indiscernible from "hardware acceleration."

CPU emulation is good and fast, and more than I need for the occasional bit of x86 compiling (native compiling is always so much easier than cross compiling). If you can ignore the massive lag of screen-scrolls, it's fine.

**Reply by pavitrakumar78** [reply]
Thanks for sharing! I've had the opposite experience with Vmware - using CMD is fine, but anything window/GUI related is so bad it affects the host. 

I'm using windows/WSL right now and wanted to shop for a laptop that can offer a similar experience. I've read about the issues with M1/M2 using dev for x86, but it has not improved so far from what I've read (maybe I'm wrong). This is pushing me to look at T-series from Lenovo or some of the newer ASUS ones with the efficient Ryzen cores.
The newer Macbooks would have been perfect if I can just port by x86_64 dev env without issues (even small performance hit would be acceptable) but it doesn't seem to be possible now.

**Reply by nonarkitten** [reply]
I don't want to OT this thread too much, but I dust-binned the Zephyrus because I couldn't stand the noise. Open Eclipse and fans kick on. Shadow copy in the background? Fans. Defender doing an AV scan? Fans. They're fast machines, but the TDP is unacceptable.

The M1 is ridiculously fast for coding because of the insane SSD to memory bandwidth available. It can be a "bit" of work setting up a proper cross compiler, but it's worth it. I can compile with GCC faster than anything else I've used, including my old ThreadRipper (which I sold after getting the M1 Air).

Running x86 Ubuntu is pretty nice. I know some people don't like "having to" run a GUI, but with HW acceleration, its a lot better. I think the text mode is a proper bug and should be easy to fix. If I had to guess, it's some bit of "premature optimization" trying to fight with sudden mass update when you get some 100-line spam from 'apt update.'

**Reply by Hamath-Eltrael** [reply]
https://github.com/utmapp/UTM/assets/60324384/030dc6b5-9215-4d9f-8f81-3695322418a2

Does it Look like this? because this is Really bad!

**Reply by Hamath-Eltrael** [reply]
M1 Pro, 4gb RAM and 4 cores

**Reply by xybytz** [reply]
That's what it looks like for me.

**Reply by Hamath-Eltrael** [reply]
Ok, so I'm not alone at leat. But I would not call this ok or even functioning. It's probably not doable better(atm) so I'll let it slide, but I would have loved and **"extremely"** slow and I would not understand this level of laggynes under:

> Run Intel/AMD systems like Windows® 7, Windows® XP, Ubuntu® Linux and many more, with reasonable speed (here we have to resort to emulation on Macs with an Apple chip - on Macs with an Intel chip, however, these systems are the fastest! )

-Appstore of UTM in German, translated by "google translate"

So either change the description to make it more obvious how bad it'll be, or just show a video of the performance when emulating a x86 on ARM, labeled CLEARLY.

**Reply by xybytz** [reply]
Maybe we should try VirtualBox or something. Will report speed.

**Reply by xybytz** [reply]
Or use Debian arm or ubuntu arm

**Reply by Hamath-Eltrael** [reply]
Of course, if it only ran on ARM. Only x68 sadly. That's what I need UTM for.

**Comment by smoe** [comment]
4.1.0 and x86_64 (Debian stable) on an M1 (OS X 13.0.1) is still slow - but it works, just unusable. I also tried an Debian ARM and that was very fast, though. Very excited.  Just, any docker instance requesting linux/amd64 then fails with what is linux/arm64.

**Comment by pavitrakumar78** [comment]
Is there still no way to reliably develop applications for x86_64 Linux servers on M1/2/X chips?

**Reply by thedocbwarren** [reply]
you can easily cross compile with gcc.  Are you writing AMD64 asm?  If you are talking about Windows, Windows 11 ARM is able to run apps for AMD64 and x86.  Linux ARM runs well on M1/2 Macs in a VM.  You can code there.  I do all the time.

**Reply by pavitrakumar78** [reply]
The servers are AMD64. I'm just writing a lot of DB-intensive applications w/ python + C-based libraries like numpy and I've seen a lot of arch. dependent dependencies for these libraries so I'm not sure if I can be 100% confident about the replicability of code behaviour when written on say, M1 and then deployed on an AMD64. 

For this usecase, the only viable solution right now is running a windows ARM within M1 so that you get as close to the x86 linux as possible?

**Comment by thedocbwarren** [comment]
It sounds like it's fully doable on the M1 with that stack.  I code quite a bit with high and lower level and deploy to multiple architectures.  For Linux it's more about matching system (like systemd for ex.,) and any library versions than architecture unless it's written specifically for the architecture.  But every case is different and I'm not sure you'd be able to do this in WSL on ARM Windows effectually.

**Comment by 40P3** [comment]
I was expecting a fast install and use of the win10 today on my M1 with UTM. Did install and check WinServer2k19 few months back and also ARM win10 those were okayish, I was able to do what I want (check KMS installation etc.).
But this install, win10pro x86_64 took half a day.
And it is completely useless. The user setup were looping with OOBESETTINGSMULTIPAGE error. had to try like 5-6 times until it managed to set up a user.
x4GB and 4cores it should be more than enough for a VGA desktop user experience on a bare win10pro install.
Mind you, tried the same more than a year ago with parallels trial and it was really smooth, even the non-arm version.
This emulation is broken for sure.

P.S.: No I cannot use arm, I need native win64, because Powershell features not supporting and working with arm.

**Comment by michael-scott-howard** [comment]
On my M1 Max, tried a MINT x86 emulation and with 4 cores, 12GB RAM the emulation is still in Sept, 2023 unusable . Works just at 50 baud. Using HTOP, at the Mac OS level, UTM using just 1 of the systems 9 actually cores...

**Reply by xybytz** [reply]
I tried windows 7 and it worked fine for me.

**Reply by rkshthrmsh** [reply]
@michael-scott-howard, found this article about bumping up the number of cores. Haven't tried it yet, but I am also trying to get a solution for this issue.

https://forums.macrumors.com/threads/utm-running-x86-64-ubuntu-20-04-on-m1-and-the-potential-for-x86-windows.2292047/

**Comment by Github-Camper** [comment]
Windows 7 and 8.1 Build 9299 is not slow, but actually fast in my Macbook Air 2021 M1

**Reply by n10v** [reply]
Did you mean MBA 2020 or MBA 2022? I can't find MBA 2021 🤔

**Reply by Github-Camper** [reply]
oops i meant 2020, 512 storage M1

**Reply by n10v** [reply]
Last time I tried about 3 months ago on my MBA 2020 M1 with 256 GB storage and I could not install Win7, because even the installer was incredibly slow

**Comment by yakimant** [comment]
Both Ubuntu 20.04 and 22.04 (x86, emulation) run well Apple M2 (Macbook Air), UTM 4.45!

Here are the settings:
- 4 cores, force multiple cores
- 4Gb RAM
- X11: Switch to LXQT and Openbox
- Wayland: Sway works well with `GPU Accelarateion Supported` Display Card.

That's it!

UPDATE: in order to make Wayland (Sway) to work properly (without glitches and fast) I had to switch video to one of the `-gl` options. Bothe `virtio-gpu-gl-pci` and `virtio-ramfb-gl` work well.

**Reply by Firdausfatihul** [reply]
explain how to switch to LXQT and OpenBox

**Reply by yakimant** [reply]
`sudo apt install lxqt openbox` + switch session
And then set wm to openbox.

**Comment by Archie-McConchie** [comment]
I think it'll always be slow when emulating it. I have 16gbs of ram on my Mac (m2) and it is still totally unusable. However, I'm sure if you download the arm version of windows 11/10 then you'll have a much smoother time.

**Comment by ccalvocm** [comment]
Greetings @eziologico I encountered the same issue in the past, but I resolved it by virtualizing Windows 11 ARM under UTM, which runs x86_64 apps seamlessly and supports GPU acceleration perfectly. I found that neither Windows 10/11 x86_64 nor Windows 10 ARM could run x86_64 apps.

**Comment by bferguson3** [comment]
Ubuntu 14 image from the UTM gallery is still completely unusable. No graphical acceleration or improvements to emulation appear to have been made in the last 4 years.

**Comment by naisanzaa** [comment]
Wow what a bummer

**Comment by jeroendiederen** [comment]
Try an unofficial linux mint for arm64. Google MintPPC and linux mint arm64. It is reported to work very well in UTM and Parallels.

**Comment by nonarkitten** [comment]
There has been some progress on making an accelerated driver for Windows 7. It presently only runs on Linux and is incredibly buggy right now, but it's moving.

HOWEVER, I found that the following minor tweaks makes Windows 7 much more useable.
- Set the number of Cores to 4
- Force Multicore (yes, it is a little unstable -- save often)
- Give yourself plenty of free RAM
- Set the JIT cache to equal to your RAM

There's also the dozens of normal tweaks for Windows 7 that help. Disable defrag and prefetch (all systems now are SSD). Turn down visual styles. Disable unnecessary services.
